packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts

```diff
@@ -58,10 +58,47 @@ export async function sendSignedXml({
     timeout: 120000,
     headers: {
       "Content-Type": "application/soap+xml;charset=UTF-8",
+      Accept: "application/soap+xml",
       "Cache-Control": "no-cache",
     },
     httpsAgent: agent,
   });
 
-  return response.data;
+  return { response: response.data, contentType: response.headers["content-type"] };
+}
+
+export async function sendSignedXmlMTOM({
+  signedXml,
+  url,
+  samlCertsAndKeys,
+  trustedKeyStore,
+}: {
+  signedXml: string;
+  url: string;
+  samlCertsAndKeys: SamlCertsAndKeys;
+  trustedKeyStore: string;
+}): Promise<{ response: string; contentType: string }> {
+  const boundary = "MIMEBoundary782a6cafc4cf4aab9dbf291522804454";
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+    cert: samlCertsAndKeys.certChain,
+    key: samlCertsAndKeys.privateKey,
+    passphrase: samlCertsAndKeys.privateKeyPassword,
+    ca: trustedKeyStore,
+    ciphers: "DEFAULT:!DH",
+    secureOptions: constants.SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION,
+  });
+
+  const payload = `--${boundary}\r\nContent-ID: <doc0@metriport.com>\r\nContent-Type: application/xop+xml; charset=UTF-8; type="application/soap+xml"\r\nContent-Transfer-Encoding: 8bit\r\n\r\n${signedXml}\r\n\r\n--${boundary}--`;
```

Comment:
constructing mtom payload on requests

Reply:
we can always send the request with the same exact MTOM headers.

Reply:
Can we create a function to build a MTOM payload? It would also be good to unit test it.

Reply:
Also, is `doc0@metriport.com` supposed to work? If so, do we have that account/alias?

Regardless, it would be good to have it as a const - we also have it below, so those would point to the same const.

Reply:
I will create a function for both the content type and payload
doc0metriport.com works. The contentId could be anything as long as its the same in the content type and body of the message.